src/SoccerStatistics.Api.Database/DataInitializer.cs

```diff
@@ -0,0 +1,317 @@
+        private static uint teamId = 1;
+        private static uint transferId = 1;
+        private static uint activityId = 1;
+        private static uint interactionBetweenPlayersId = 1;
+        private static uint teamInMatchStatsId = 1;
+        private static uint matchId = 1;
+        private static uint roundId = 1;
+        private static uint teamInLeagueId = 1;
+        private static uint leagueId = 1;
+        #endregion
+
+        private readonly SoccerStatisticsDbContext _context;
+        private readonly ILogger<DataInitializer> _logger;
+
+        public DataInitializer(SoccerStatisticsDbContext context, ILogger<DataInitializer> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        public void Seed()
+        {
+            const int teamCount = 5;
+            const int transferCount = 50;
+            const int leagueCount = 5;
+
+            var teams = GetFakeTeam().Generate(teamCount);
+            var players = teams.SelectMany(x => x.Players);
+
+            _context.Stadiums.AddRange(teams.Select(x => x.Stadium));
+            _logger.LogTrace("Stadiums are added");
+
+            _context.Players.AddRange(players);
+            _logger.LogTrace("Players are added");
+
+            _context.Teams.AddRange(teams);
+            _logger.LogTrace("Teams are added");
+
+            _context.Transfers.AddRange(GetFakeTransfer(players, teams).Generate(transferCount));
+            _logger.LogTrace("Transfers are added");
+
+            var leagues = GetFakeLeague(teams).Generate(leagueCount);
+
+            _context.Leagues.AddRange(leagues);
+            _logger.LogTrace("Leagues are added");
+
+            _context.Rounds.AddRange(leagues.SelectMany(x => x.Rounds));
+            _logger.LogTrace("Rounds are added");
+
+            var matches = leagues.SelectMany(x => x.Rounds.SelectMany(y => y.Matches));
+
+            _context.Teams_in_match_stats.AddRange(matches.SelectMany(x => new List<TeamInMatchStats> { x.TeamOneStats, x.TeamTwoStats }));
+            _logger.LogTrace("Team stats in matches are added");
+
+            _context.Matches.AddRange(matches);
+            _logger.LogTrace("Matches are added");
+
+            _context.Activities.AddRange(matches.SelectMany(x => x.Activities));
+            _logger.LogTrace("Activities are added");
+
+            _context.Interactions_between_players.AddRange(matches.SelectMany(x => x.InteractionsBetweenPlayers));
+            _logger.LogTrace("Interactions are added");
+
+            try
+            {
+                _logger.LogTrace("All data are saving in the database");
+                _context.SaveChanges();
+            }
+            catch (Exception err)
+            {
+                _logger.LogError($"Exception message: {err.Message}");
+                _logger.LogError($"InnerException message: {err.InnerException.Message}");
+
+                _logger.LogWarning("Data aren't saved");
```

Comment:
Sounds funny https://language-boutique.com/lost-in-translation-full-reader/these-data-are-or-this-data-is-.html